docs/source/index.rst

```diff
@@ -1,55 +1,121 @@
-Knit enables users to express and deploy applications and software environments managed under YARN through Python.
+Knit provides Python methods to quickly launch, monitor, and destroy
+distributed programs running on a YARN cluster, such as is found in traditional
+Hadoop environments.
 
-Knit is part of the `Dask`_ project, committed to bringing cluster-based data science within easy
-reach of python developers.
+Knit was originally designed to deploy Dask_ applications on YARN, but can
+deploy more general, non-Dask, applications as well.
 
-Motivation
-----------
+Using conda_, Knit can also deploy fully-featured Python environments within
+YARN containers, sending along useful libraries like NumPy, Pandas, and
+Scikit-Learn to all of the containers in the YARN cluster.
 
-Knit was built to support batch-oriented non-JVM applications.  For example, Knit can deploy Python based distributed
-applications such as `IPython Parallel`_, with particular support for `Dask`_.
-Knit was built with the following motivations in
-mind:
+Install
+-------
 
-*  **PyData Support** Bring the PyData stack into the Hadoop/YARN ecosystem
-*  **Easy Setup:** Support a minimal installation effort and the common cases with easy to use Python interface.
-*  **Deployable Runtimes:** Build and ship self contained environments along with the application.  Knit uses `conda`_
-   to resolve library dependencies and deploy user libraries without IT infrastructure and management
+Use ``pip`` or ``conda`` to install::
+
+   conda install knit -c conda-forge
+   # or
+   pip install knit
+
+
+Launch Generic Applications
+---------------------------
+
+Instantiate ``knit`` with valid ResourceManager/Namenode IP/Ports and create a command string to run
+in all YARN containers
+
+.. code-block:: python
+
+   from knit import Knit
+   k = Knit(autodetect=True) # autodetect IP/Ports for YARN/HADOOP
+
+Create a software environment with necessary packages
+
+.. code-block:: python
+
+   env = k.create_env('my-environment',
+                      packages=['python=3.5', 'scikit-learn','pandas'],
+                      channels=['conda-forge'])  # specify anaconda.org channels
+
+Run a command within that environment on multiple containers
+
+.. code-block:: python
+
+   cmd = 'python -c "import sys; print(sys.version_info);"'
+   app_id = k.start(cmd, num_containers=2, env=env)  # start application
+
+The ``start`` method also takes parameters to define the resource requirements
+of the application like ``num_containers=``, ``memory=``, ``virtual_cores=``,
+``env=``, and ``files=``.
+
+Launch Dask
+-----------
+
+Knit makes it easy to launch Dask on Yarn:
+
+.. code-block:: python
+
+   from knit.dask_yarn import DaskYARNCluster
+   env = k.create_env('my-environment',
+                      packages=['python=3.6', 'scikit-learn', 'pandas', 'dask'],
+                      channels=['conda-forge'])  # specify anaconda.org channels
+
+   cluster = DaskYARNCluster(env=env)
```

Comment:
You can specify channels and packages in the constructor here, and then you don't need the lines above, or to instantiate a Knit at all. The environment gets a hashed name.

Reply:
Given that we talk about it above I thought I would reuse the same pattern.  Happy either way though.

Reply:
It's good this way too, just wanted to be sure. 
Another section that could be included is instrance structure, since I don't think it's too clear right now. It's not exactly complicted, but parameters do get passed all the way down. These are instance attributes:

    DaskYarnCluster
        - Dask LocalCluster, scheduler only
        - Knit
            - YARNApi
            - CondaCreator
            - HDFileSystem